src/conversion/upcastdispatcher.js

```diff
@@ -301,25 +304,64 @@ export default class UpcastDispatcher {
 	}
 
 	/**
-	 * Checks if {@link #_removeIfEmpty} contains empty elements and remove them.
-	 * We need to do it smart because there could be elements that are not empty because contains
-	 * other empty elements and after removing its children they become available to remove.
-	 * We need to continue iterating over split elements as long as any element will be removed.
+	 * Registers that `split` element is a split part of the `original` element.
+	 *
+	 * Data set by this method is then used by {@link #_getSplitParts} and {@link #_removeEmptyElements}
+	 *
+	 * @private
+	 * @param {module:engine/model/element~Element} original The original element.
+	 * @param {module:engine/model/element~Element} split The split part element.
+	 */
+	_registerSplitPair( original, split ) {
+		if ( !this._splitParts.has( original ) ) {
+			this._splitParts.set( original, [ original ] );
+		}
+
+		const list = this._splitParts.get( original );
+
+		this._splitParts.set( split, list );
+		list.push( split );
+	}
+
+	/**
+	 * @private
+	 * @see module:engine/conversion/upcastdispatcher~UpcastConversionApi#getSplitParts
+	 */
+	_getSplitParts( element ) {
+		let parts;
+
+		if ( !this._splitParts.has( element ) ) {
+			parts = [ element ];
+		} else {
+			parts = this._splitParts.get( element );
+		}
+
+		parts.last = parts[ parts.length - 1 ];
```

Comment:
Actually, I am not sure if we should do it. We do not do it for any other array, even considering that in many cases it could be useful. If we will do it now, we may do it more often and end up in the situation where half of arrays has property `last` and the other half do not.

Reply:
That was your idea :P But, yeah, I am not sure about it either.

Reply:
`if (!Array.prototype.last) { Object.defineProperty...` 😈 😄

Reply:
:trollface: